### DISCUSSION TITLE
packaging/overview

### DISCUSSION THREAD

**Discussion by giscus** [velopack/velopack.docs#7] [General]
# packaging/overview

Packaging a release is accomplished with the pack command in Velopack. Regardless of your operating system, the common required arguments are roughly the same.

https://docs.velopack.io/packaging/overview

**Comment by sameerdube9** [comment]
How do I add other dependent files such as ffmpeg.exe required for application?

**Reply by caesay** [reply]
You should copy any files you need into your app pack directory before calling `vpk pack`. Usually this can be done automatically during your compile (eg. use [`CopyToOutputDirectory`](https://stackoverflow.com/questions/44374074/copy-files-to-output-directory-using-csproj-dotnetcore)).

**Comment by cesar-albusac** [comment]
Do you know if it's possible to do use velopack using .msi rather than  .exe? I get this error: 

`-e does not have an .exe extension` 

But not sure if there are a workaround. 

Thanks!

**Reply by caesay** [reply]
The file you use with -e must be the entry point / main exe for your application, and that entry point must implement VelopackApp builder. It would not make sense for you to pass an msi, that is not a binary nor can it implement velopack.

Velopack can not yet create msi either, but it is on our backlog.

**Comment by cesar-albusac** [comment]
Is there a way to avoid this question or make it default to yes? 

`A release in this channel with the same or greater version already exists. Do you want to continue and potentially
overwrite files? [y/n] (n):`

I have releases for version 3, version 4 , version 5. And they are being created independly, so maybe yesterday I created 5.4.32 and today I am creating 3.8.4 , and I have integrated on a Gitlab Pipeline, so I would like to create it without the question. 

Or is it better to keep all those 3 with different --outputDir ?

**Reply by caesay** [reply]
As I answered on your other post, you probably should be using channels for this. Velopack does not support building releases out of order on any given channel, it expects that the new version will always be greater than the one before it. If you use channels, you can separate each one of these (eg. imagine a `v3` channel that you use to publish v3 updates, and a `v5` channel for v5 updates etc). 

But to answer your actual question, you can turn on non-interactive mode with `-x` and you can default to yes with `-y`. This is shown in the help docs available with `vpk -h`.

**Comment by NeighborhoodNerd** [comment]
Is there any way to disable Velopack from running post dotnet publish? Currently, when I run dotnet publish, it packs the release, then I have to do it again so that I can add code signing to the arguments.

**Reply by NeighborhoodNerd** [reply]
Figured it out, had to remove the Velopack.build nuget from the csproj.
